apiserver/common/networkingcommon/linklayer.go

```diff
@@ -232,32 +235,34 @@ func (o *MachineLinkLayerOp) AssertAliveOp() txn.Op {
 
 // MarkDevProcessed indicates that the input hardware address was present in
 // the incoming data and its updates have been handled by the build step.
-func (o *MachineLinkLayerOp) MarkDevProcessed(hwAddr string) {
-	o.processedDevs.Add(hwAddr)
+func (o *MachineLinkLayerOp) MarkDevProcessed(name, hwAddr string) {
+	o.processedDevs.Add(name + hwAddr)
```

Comment:
I wonder if we should have a function for creating the unique namespace? That way we can reuse it for all locations (i.e. the repetitions below)

Reply:
Added a key function.